--- a/riscv/processor.cc
+++ b/riscv/processor.cc
@@ -267,7 +267,9 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
     }
   }
-  csrmap[CSR_MCOUNTINHIBIT] = std::make_shared<const_csr_t>(proc, CSR_MCOUNTINHIBIT, 0);
+  // NEMU DiffTest: enable Zicntr mcountinhibit CY/IR.
+  csrmap[CSR_MCOUNTINHIBIT] =
+    std::make_shared<masked_csr_t>(proc, CSR_MCOUNTINHIBIT, 0x5, 0);
   if (proc->extension_enabled_const(EXT_SSCOFPMF))
     csrmap[CSR_SCOUNTOVF] = std::make_shared<scountovf_csr_t>(proc, CSR_SCOUNTOVF);
   csrmap[CSR_MIE] = mie = std::make_shared<mie_csr_t>(proc, CSR_MIE);
